### DISCUSSION TITLE
How to convert outputted dicts to srt in Python?

### DISCUSSION THREAD

**Discussion by FilipRank** [openai/whisper#1904] [Show and tell]
There's obviously a way to do it, since using whisperAI through the CLI outputs this filetype, but I can't find any documentation for WhisperAI except for the Read_me so I'm asking here instead.

**Comment by ryanheise** [comment]
Try typing the keywords "python srt" from the Discussions page as shown below:

![Screenshot 2023-12-17 at 9 42 39 pm](https://github.com/openai/whisper/assets/19899190/38ecf59d-17f8-4b92-99a8-d6e055807621)

The 3rd link has an answer. (The top 2 point to the 3rd.)

**Comment by StarMariner** [comment]
I asked Bing chat for you and it spat this out: May be of use? 
````
def dict_to_srt(subtitles_dict):
    srt_content = ""
    for index, sub_data in enumerate(subtitles_dict, start=1):
        start_time = sub_data['start']
        end_time = sub_data['end']
        text = sub_data['text']
        srt_content += f"{index}\n{start_time} --> {end_time}\n{text}\n\n"
    return srt_content
```

# Example usage:
subtitles_dict = [
    {'start': '00:00:01,000', 'end': '00:00:03,000', 'text': 'First subtitle text.'},
    {'start': '00:00:04,000', 'end': '00:00:06,000', 'text': 'Second subtitle text.'}
]

srt_output = dict_to_srt(subtitles_dict)
with open('output.srt', 'w') as file:
    file.write(srt_output)`

**Reply by ryanheise** [reply]
> I asked Bing chat for you and it spat this out: May be of use?

Without going into the details of why, no Bing Chat did not spit out the correct answer. However, there is no need to contribute another answer since this question has already been asked and answered before (with the correct answer), and you can see the previous question and answer by a search as illustrated above.
